__tests__/app.test.js

```diff
@@ -391,3 +401,40 @@ describe("GET /api/users", () => {
       });
   });
 });
+describe("GET /api/articles by query", () => {
+  test("returns all articles with specified topic", () => {
+    return request(app)
+      .get("/api/articles/?topic=cats")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        console.log(articles);
+        expect(articles.length).toBe(1);
+        expect(typeof articles[0].author).toBe("string");
+        expect(typeof articles[0].title).toBe("string");
+        expect(typeof articles[0].article_id).toBe("number");
+        expect(typeof articles[0].topic).toBe("string");
+        expect(typeof articles[0].created_at).toBe("string");
+        expect(typeof articles[0].votes).toBe("number");
+        expect(typeof articles[0].article_img_url).toBe("string");
+        expect(typeof articles[0].comment_count).toBe("string");
+      });
+  });
+  test("returns all articles if passed no query", () => {
+    return request(app)
+      .get("/api/articles/?topic=")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        expect(articles.length).toBe(13);
+      });
+  });
+  test("non-existent query returns status 404 and error message", () => {
+    return request(app)
+      .get("/api/articles/?topic=youguessedit-bananas")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("not found");
+      });
+  });
```

Comment:
Can you add another 200 test where you check if you look for articles with a valid topic, but it has no articles, you get an empty array back 🔍